spec/app/external_doc_spec.rb

```diff
@@ -33,15 +44,31 @@
         expect(html).not_to have_selector("h1", text: "Lorem ipsum")
       end
 
-      it "rewrites links to markdown pages" do
-        expect(html).to have_link("turpis ultrices", href: "/ultrices.html")
-      end
-
       it "does not rewrite links to markdown pages with a host" do
         expect(html).to have_link("Nam eget dui", href: "https://nam.com/eget/dui/uploading.md")
         expect(html).not_to have_link("Nam eget dui", href: "https://nam.com/eget/dui/uploading.html")
       end
 
+      it "converts GitHub relative links to absolute links if link is outside of the `docs` folder" do
+        expect(html).to have_link("turpis ultrices", href: "https://github.com/alphagov/lipsum/blob/master/ultrices.md")
+      end
+
+      context "the repository has not set its `consume_docs_folder` property" do
+        it "converts GitHub relative links to absolute links even if link is in the `docs` folder" do
+          expect(html).to have_link("Read the docs", href: "https://github.com/alphagov/lipsum/blob/master/docs/foo.md")
+          expect(html).to have_link("Read another doc", href: "https://github.com/alphagov/lipsum/blob/master/docs/bar.md")
+        end
+      end
+
+      context "the repository is set to `consume_docs_folder: true`" do
+        let(:consume_docs_folder) { true }
+
+        it "rewrites GitHub relative links to Developer Docs HTML links if link is in the `docs` folder" do
+          expect(html).to have_link("Read the docs", href: "/apis/lipsum/foo.html")
+          expect(html).to have_link("Read another doc", href: "/apis/lipsum/bar.html")
+        end
+      end
```

Comment:
These tests could be written more cleanly, with clearer contexts. Also missing explicit tests for when we're parsing something as a `docs` (rather than as root).